tests/functests/instancetype_test.go

```diff
@@ -0,0 +1,131 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/utils/pointer"
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+)
+
+var _ = Describe("Common instance types func tests", func() {
+
+	Context("VirtualMachine using a preference with resource requirements", func() {
+		It("is rejected if it does not provide enough resources", func() {
+			createInstancetype(2, "tiny-instancetype", "64M")
+			intanceTypeMatcher := v1.InstancetypeMatcher{
+				Name: "tiny-instancetype",
+				Kind: "VirtualMachineInstancetype",
+			}
+			vm := randomVMWithInstancetype(&intanceTypeMatcher)
+
+			clusterPreferences, err := virtClient.VirtualMachineClusterPreference().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(clusterPreferences.Items).ToNot(BeEmpty())
+
+			for _, preference := range clusterPreferences.Items {
+				vm.Name = fmt.Sprintf("test-vm-%s", preference.Name)
+				vm.Spec.Preference = &v1.PreferenceMatcher{
+					Name: preference.Name,
+				}
+
+				_, err := virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+				Expect(err).To(MatchError(
+					fmt.Sprintf(
+						"admission webhook \"virtualmachine-validator.kubevirt.io\" denied the request: "+
+							"failure checking preference requirements: "+
+							"insufficient Memory resources of 64M provided by instance type, preference requires %s",
+						preference.Spec.Requirements.Memory.Guest.String(),
+					)))
+			}
+		})
+
+		It("can be created when enough resources are provided", func() {
+			intanceTypeMatcher := v1.InstancetypeMatcher{
+				Name: "u1.large",
+			}
+			vm := randomVMWithInstancetype(&intanceTypeMatcher)
+
+			clusterPreferences, err := virtClient.VirtualMachineClusterPreference().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(clusterPreferences.Items).ToNot(BeEmpty())
+
+			for _, preference := range clusterPreferences.Items {
+				vm.Name = fmt.Sprintf("test-vm-%s", preference.Name)
+				vm.Spec.Preference = &v1.PreferenceMatcher{
+					Name: preference.Name,
+				}
+
+				vm, err := virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func(g Gomega) {
+					vm, err := virtClient.VirtualMachine(testNamespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(vm.Status.Ready).To(BeTrue())
+				}, 60*time.Second, time.Second).Should(Succeed())
```

Comment:
Can you make the timeout a constant at the beginning of the file and set it to a higher value like e.g. `300*time.Second` please?